FileShareService.DesktopClient.Core/JobsParser.cs

```diff
@@ -26,12 +35,144 @@ public Jobs.Jobs Parse(string jobs)
+
+                ErrorDeserializingJobsJob? errorJob = null;
+
+                // Deserialize each job and validate
+                foreach (JToken batchJob in batchJobs)
+                {
+                    string jobAction = string.Empty;
+
+                    List<string> errors = ValidateJobActionAndDisplayName(batchJob, out jobAction);
+
+                    if(errors.Any())
+                    {
+                        if(errorJob == null)
+                        {
+                            errorJob = new ErrorDeserializingJobsJob();
+                        }
+                        errorJob.ErrorMessages.AddRange(errors);
+
+                        continue;
+                    }
+
+                    IJob? job = null;
+                    string jsonString = Convert.ToString(batchJob);
+
+                    switch (jobAction)
+                    {
+                        case NewBatchJob.JobAction:
+                            job = JsonConvert.DeserializeObject<NewBatchJob>(jsonString, jsonSerializerSettings);
+                            break;
+
+                        case AppendAclJob.JobAction:
+                            job = JsonConvert.DeserializeObject<AppendAclJob>(jsonString, jsonSerializerSettings);
+                            break;
+
+                        case SetExpiryDateJob.JobAction:
+                            job = JsonConvert.DeserializeObject<SetExpiryDateJob>(jsonString, jsonSerializerSettings);
+                            break;
+                    }
+
+                    if (job != null)
+                    {
+                        job.Validate(batchJob);
+                        jobCollection.Add(job);
+                    }
+                }
+
+                //Add error job at begining of the collection, if exists.
+                if (errorJob != null)
+                {
+                    jobCollection.Insert(0, errorJob);
+                }
+
+                return  new Jobs.Jobs() { jobs = jobCollection};
             }
             catch (Exception e)
             {
-                return new Jobs.Jobs() {jobs = new[] {new ErrorDeserializingJobsJob(e)}};
+                return new Jobs.Jobs() { jobs = new[] { new ErrorDeserializingJobsJob(e) { ErrorMessages = new List<string>() { e.Message } } } };
+            }
+        }
+
+        private List<string> ValidateJobActionAndDisplayName(JToken job, out string jobAction)
+        {
+            jobAction = string.Empty;
+
+            List<string> errors = new List<string>();
+
+            //Retrieve job action
+            JToken? jobActionToken = job.SelectToken("action");
+
+            if (jobActionToken == null ||
+                string.IsNullOrWhiteSpace(Convert.ToString(jobActionToken)))
+            {
+                errors.Add(AddLineInfo(job, "Job action is not specified or is invalid."));
+                return errors;
             }
+
+            jobAction = $"{Convert.ToString(jobActionToken)}";
```

Comment:
String interpolation isn't needed here. Please change to
````
jobAction = Convert.ToString(jobActionToken);
````

Reply:
Fixed